core/db_models/EEM_Transaction.model.php

```diff
@@ -380,18 +380,18 @@ public function delete_junk_transactions()
         if (! empty($txn_ids) && is_array($txn_ids)) {
             // first, make sure these TXN's are removed the "ee_locked_transactions" array
             EEM_Transaction::unset_locked_transactions($txn_ids);
+
+            // Create an array of sanitized numbers.
+            $safe_txn_ids = array();
+            
+            foreach ($txn_ids as $id) {
+                $safe_txn_ids[] = intval($id);
+            }
+
             // let's get deletin'...
-            // Why no wpdb->prepare?  Because the data is trusted.
             // We got the ids from the original query to get them FROM
             // the db (which is sanitized) so no need to prepare them again.
-            $query   = $wpdb->prepare('
-				DELETE
-				FROM %s
-				WHERE
-                    TXN_ID IN ( %s )',
-                $this->table(),
-                implode(",", $txn_ids)
-            );
+            $query   = $wpdb->prepare('DELETE FROM %s WHERE TXN_ID IN ( %s )', $this->table(), implode(',', $safe_txn_ids));
```

Comment:
this still won't work with `$wpdb->prepare` because it is still declaring a placeholder for a string:

```php
WHERE TXN_ID IN ( %s )
```

which will result in the actual query looking something like:

```php
WHERE TXN_ID IN ( '1,2,3,4' )
```

whereas we need it to look something like:

```php
WHERE TXN_ID IN ( 1,2,3,4 )
```

notice the latter version does not have quotes around the TXN IDs, which means the database will now interpret those as numbers instead of a single string like `'1,2,3,4'`.

so you need to change how the placeholders are generated so that it produces something like:

```php
WHERE TXN_ID IN ( %d,%d,%d,%d )
```

Now that said, running all of the TXN IDs through `intval()` and then imploding the `$safe_txn_ids` array in the query is appropriate sanitization but just won't work with  `$wpdb->prepare`. So you could revert back to the original code that did not prepare the query and use your `$safe_txn_ids` idea instead if you like, or properly prepare the placeholders like I have previously described.

Your choice.